src/TableResizePlugin.js

```diff
@@ -0,0 +1,568 @@
+      );
+    }
+    return state;
+  }
+}
+
+// Function that handles the mousemove event inside table cell.
+function handleMouseMove(view: EditorView, event: PointerEvent): void {
+  const resizeState = PLUGIN_KEY.getState(view.state);
+  if (resizeState.draggingInfo) {
+    return;
+  }
+
+  const target = domCellAround(event.target);
+  let forMarginLeft = false;
+  let cell = -1;
+
+  if (target instanceof HTMLElement) {
+    const {left, right} = target.getBoundingClientRect();
+    const offsetLeft = event.clientX - left;
+    if (offsetLeft <= HANDLE_WIDTH) {
+      if (target.cellIndex === 0) {
+        forMarginLeft = true;
+        cell = edgeCell(view, event, 'right');
+      } else {
+        cell = edgeCell(view, event, 'left');
+      }
+    } else if (right - event.clientX <= HANDLE_WIDTH) {
+      cell = edgeCell(view, event, 'right');
+    }
+  }
+
+  if (
+    cell === resizeState.cellPos &&
+    forMarginLeft === resizeState.forMarginLeft
+  ) {
+    return;
+  }
+
+  if (cell !== -1) {
+    const $cell = view.state.doc.resolve(cell);
+    if (!$cell) {
+      return;
+    }
+  }
+  updateResizeHandle(view, cell, forMarginLeft);
+}
+
+// Function that handles the mouseleave event from the table cell.
+function handleMouseLeave(view: EditorView): void {
+  const resizeState = PLUGIN_KEY.getState(view.state);
+  const {cellPos, draggingInfo} = resizeState;
+  if (cellPos > -1 && !draggingInfo) {
+    updateResizeHandle(view, -1, false);
+  }
+}
+
+// Function that handles the mousedown event from the table cell.
+function handleMouseDown(view: EditorView, event: MouseEvent): boolean {
+  const resizeState = PLUGIN_KEY.getState(view.state);
+  if (resizeState.cellPos === -1 || resizeState.draggingInfo) {
+    return false;
+  }
+  view.dispatch(
+    view.state.tr.setMeta(PLUGIN_KEY, {
+      setDraggingInfo: calculateDraggingInfo(view, event, resizeState),
+    })
+  );
+
+  // Move events should be batched to avoid over-handling the mouse
+  // event.
+  const dragMove = batchMouseHandler(handleDragMove);
+
+  const finish = (event: MouseEvent) => {
+    window.removeEventListener('mouseup', finish, true);
+    window.removeEventListener('mousemove', move, true);
+    handleDragEnd(view, event);
+  };
+
+  const move = (event: MouseEvent) => {
+    if (event.which) {
+      dragMove(view, event);
+    } else {
+      finish(event);
+    }
+  };
+
+  window.addEventListener('mouseup', finish, true);
+  window.addEventListener('mousemove', move, true);
+  event.preventDefault();
+  return true;
+}
+
+// Function that handles the mouse event while resizing the table cell.
+// This will temporarily updates the table's style until the resize ends.
+function handleDragMove(view: EditorView, event: PointerEvent): void {
+  const resizeState = PLUGIN_KEY.getState(view.state);
+  const {draggingInfo, forMarginLeft} = resizeState;
+  if (!draggingInfo) {
+    return;
+  }
+  const {
+    startX,
+    columnWidths,
+    columnIndex,
```

Comment:
here and in `calculateDraggingInfo`, rename this to `targetColumn` or something like that